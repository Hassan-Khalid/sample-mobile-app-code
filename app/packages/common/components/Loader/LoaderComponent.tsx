import React from 'react';
import { ActivityIndicator, View } from 'react-native';
import styles from './styles/LoaderComponent.styles';
import APP_COLORS from '../../assets/colors';
import componentBuilder from 'common/builder/ComponentBuilder';

function LoaderComponent({ loading }: { loading: boolean }) {
  return loading ? (
    <View style={styles.spinnerWrapper} {...componentBuilder.createTestAttributes('loader-component')}>
      <ActivityIndicator size="large" color={APP_COLORS.GRAY1} />
    </View>
  ) : (
    <View />
  );
}

export default LoaderComponent;
