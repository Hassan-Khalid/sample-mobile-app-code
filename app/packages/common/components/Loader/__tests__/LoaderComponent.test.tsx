import React from 'react';
import { render } from '@testing-library/react-native';
import LoaderComponent from '../LoaderComponent';

describe('LoaderComponent', () => {
  it('renders correctly', () => {
    // exercise
    const loading = true;
    const wrapper = render(<LoaderComponent loading={loading} />);

    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render empty', () => {
    // exercise
    const loading = false;
    const wrapper = render(<LoaderComponent loading={loading} />);

    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
