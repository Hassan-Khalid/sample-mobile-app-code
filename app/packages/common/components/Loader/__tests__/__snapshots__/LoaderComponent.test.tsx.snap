// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderComponent renders correctly 1`] = `
<View
  accessibilityLabel="loader-component"
  style={
    {
      "alignItems": "center",
      "height": 1334,
      "justifyContent": "center",
      "position": "absolute",
      "width": 750,
      "zIndex": 10000,
    }
  }
  testID="loader-component"
>
  <ActivityIndicator
    color={
      {
        "dynamic": {
          "dark": "#748094",
          "highContrastDark": undefined,
          "highContrastLight": undefined,
          "light": "#8798ad",
        },
      }
    }
    size="large"
  />
</View>
`;

exports[`LoaderComponent should render empty 1`] = `<View />`;
